tests/test_utils_gz.py

```diff
@@ -25,5 +25,10 @@ def test_gunzip_no_gzip_file_raises(self):
 
     def test_gunzip_truncated_short(self):
         with open(join(SAMPLEDIR, 'truncated-crc-error-short.gz'), 'rb') as f:
-            text = gunzip(f.read())
+            text = gunzip(f.read()) 
             assert text.endswith(b'</html>')
+
+    def test_gunzip_illegal_eof(self):
+        with open(join(SAMPLEDIR, 'unexpected-eof.gz'), 'rb') as f:
+            text = gunzip(f.read())
+            self.assertEqual(text.count(b'document'), 1)
```

Comment:
This test might be a little vague. If you can, could you try creating a new gzipped file with illegal ending, and then compare the full expected text? I don't know how hard that is though, so if it takes you more than say 15 minutes just forget about it. ;)